test/api/buffer/api_test.go

```diff
@@ -372,18 +379,29 @@ func RunRequests(
 	t *testing.T,
 	envProvider testhelper.EnvProvider,
 	testDirFs filesystem.Fs,
+	workingDirFs filesystem.Fs,
 	apiUrl string,
 ) {
 	t.Helper()
 	client := resty.New()
 	client.SetBaseURL(apiUrl)
 
+	// Dump raw HTTP request
+	client.SetPreRequestHook(func(client *resty.Client, request *http.Request) error {
+		if dumpDir, ok := request.Context().Value(dumpDirCtxKey).(string); ok {
+			reqDump, err := httputil.DumpRequest(request, true)
+			assert.NoError(t, err)
+			assert.NoError(t, workingDirFs.WriteFile(filesystem.NewRawFile(filesystem.Join(dumpDir, "request.txt"), string(reqDump))))
+		}
+		return nil
+	})
+
```

Comment:
Dumpuje sa raw HTTP request to `.out`.
Je to spravene takto, lebo po requeste je `request body` uz precitane, a neda sa nacitat.